Line icons addition

MichaelMarcialis opened the pull request:
### Summary

Added new `lineDashed`, `lineDotted`, and `lineSolid` icons to EUI.

![image](https://user-images.githubusercontent.com/3884767/154120052-9e1943e3-328c-4547-a0da-e950484b6da8.png)

CCing @mbondyra, as https://github.com/elastic/kibana/pull/124970 mentions needing these icons as part of the proposed design updates to Lens flyouts.

### Checklist

- [x] Check against **all themes** for compatibility in both light and dark modes
- [x] Checked in **Chrome**, **Safari**, **Edge**, and **Firefox**
- [x] Added **[documentation](https://github.com/elastic/eui/blob/main/wiki/documentation-guidelines.md)**
- [x] Added or updated **[jest](https://github.com/elastic/eui/blob/main/wiki/testing.md) and [cypress](https://github.com/elastic/eui/blob/main/wiki/cypress-testing.md) tests**
- [x] A **[changelog](https://github.com/elastic/eui/blob/main/wiki/documentation-guidelines.md#changelog)** entry exists and is marked appropriately

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5633/

miukimiu reviewed (changes requested):
Thanks, @MichaelMarcialis for adding this new great set of icons to EUI! 🎉 

The icons look good to me. But I noticed that the `lineDotted` looks thinner when compared with the other two.

<img width="88" alt="Screenshot 2022-02-14 at 19 25 41" src="https://user-images.githubusercontent.com/2750668/153932347-a7117525-3cfc-40c1-bcf6-b0c9bd9a7f3f.png">

What do you think of instead of making a circle of `2px x 2px` making it instead with `3px x 3px`. This change makes the circles bigger with more space between them, and it's easier to notice that they are dots. 

<img width="91" alt="Screenshot 2022-02-14 at 19 25 35" src="https://user-images.githubusercontent.com/2750668/153932484-8d5779d9-c562-4a34-9004-348632a6d117.png">

I guess one of the reasons you went with the `2px x 2px` circles version is that they get better centered aligned. The  `3px x 3px` doesn't get centered properly but when is side by side with the other icons it doesn't make that much difference.

<img width="1223" alt="Screenshot 2022-02-14 at 19 29 22" src="https://user-images.githubusercontent.com/2750668/153933609-61f888c1-9d7d-4bb5-a4bd-ed6cd9c7a6c6.png">

Let me know what do you think of the proposed changes?

MichaelMarcialis commented:
> What do you think of instead of making a circle of `2px x 2px` making it instead with `3px x 3px`.

Thanks for the feedback, @miukimiu! I've updated the `dottedLine` icon to use 3x3px dots. Updated screenshot in the original description above.

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5633/

miukimiu reviewed (approved):
Thanks, @MichaelMarcialis. 

Tested in Safari, Firefox, and Chrome. Also tested in low resolution and LGTM! 🎉
